docs/explanation/charm-architecture.md

```diff
@@ -0,0 +1,23 @@
+# Charm architecture
+
+The SAML Integrator charm consists on a charm workload that fetches the SAML metadata details based on the provided configuration and propagates them through the relation.
```

Comment:
I'd suggest "The SAML Integrator charm fetches SAML metadata details based on the provided configuration and propagates them through a Juju integration." - you've got a duplication of charm midway through the sentence and we should prefer the term integration over relation where possible.

Reply:
Much better, indeed